assets/js/translator.js

```diff
@@ -979,22 +1001,22 @@ function detectLanguage() {
         error: handleDetectLanguageErrorResponse,
         complete: function () {
             ajaxComplete();
-            translateRequest = undefined;
+            translateRequest = null;
         }
     }, '/identifyLang');
 
     return translateRequest;
 
     function handleDetectLanguageSuccessResponse(data) {
-        var possibleLanguages = [];
+        var possibleLanguages /*: string[][] */ = [];
         for(var lang in data) {
             possibleLanguages.push([lang.indexOf('-') !== -1 ? lang.split('-')[0] : lang, data[lang]]);
         }
-        possibleLanguages.sort(function (a, b) {
-            return b[1] - a[1];
+        possibleLanguages.sort(function (a /*: string[] */, b /*: string[] */) /*: number */ {
+            return parseInt(b[1], 10) - parseInt(a[1], 10);
```

Comment:
Is this sorting by first character of a string instead of the entire string...? wtf

Reply:
Shall I make this so that is compares the entire string?